rust/src/adapter/mod.rs

```diff
@@ -51,10 +78,15 @@ impl Collector {
         thread::sleep(time::Duration::from_millis(50));
     }
 
-    pub async fn set_metadata(&self, key: String, value: u64) {
+    pub async fn set_metadata(&self, key: String, value: TelemetryId) {
         self.send_events
             .send(Event::Metadata(0, Metadata { key, value }))
             .await
             .unwrap();
     }
 }
+
+fn next_id() -> usize {
```

Comment:
Brought this in for sharing as well. I think it's okay to share

Reply:
I think this is fine, it's not highly contested and we shouldn't be counting high enough that it rolls over.